### DISCUSSION TITLE
Page Header

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#283] [Component suggestions]
<img width="720" alt="visuele weergave van de page header component" src="https://github.com/user-attachments/assets/31c0ab20-0e68-4f89-a610-40a524f7e256">

We willen graag meer leren over de **Page Header** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1733486460873309)

---

## Naam
Page Header

## Doel
Consistente plek bovenaan elke pagina, vaak met een logo, navigatie en zoekfunctie.

## Ook bekend als
Header

## Gerelateerde componenten
[Root](https://github.com/orgs/nl-design-system/discussions/426), [Body](https://github.com/orgs/nl-design-system/discussions/427), [Page Layout](https://github.com/orgs/nl-design-system/discussions/428), [Page Body](https://github.com/orgs/nl-design-system/discussions/429), [Page Footer](https://github.com/orgs/nl-design-system/discussions/282), [Navigation Bar](https://github.com/orgs/nl-design-system/discussions/267)

## Gerelateerde patronen
[Page Header - Meerstappenformulier](https://github.com/orgs/nl-design-system/discussions/382)

## Onderzoek

[Launching GOV.UK’s new menu bar - GOV.UK](https://insidegovuk.blog.gov.uk/2021/11/11/launching-gov-uks-new-menu-bar/)
Dit artikel gaat over verbeteringen in de navigatie op GOV.UK om overheidsinformatie en -diensten toegankelijker te maken. Het beschrijft de introductie van een consistente menu balk die gebruikers helpt informatie te vinden op basis van thema’s, organisaties en inhoudstypen, in plaats van complexe overheidsstructuren.

[GOV.UK header - GOV.UK](https://design-system.service.gov.uk/components/header/)
Header implementatie bij GOV.UK. Bij de [Header Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/97) wordt meer informatie, onderzoek en aanpassingen gedeeld. Enkele interessante comments:

- Uit een onderzoek met 50 gebruikers bleek dat gebruikers moeite hadden met het vinden van navigatie-opties. Dit kwam doordat de navigatiebalk slecht te onderscheiden was van de koptekst, door dezelfde kleur- en achtergrondstijl, en doordat de koptekst verdween bij het scrollen. Een alternatieve layout met betere zichtbaarheid en uitlijning presteerde beter.
- Overweeg om ‘show or hide’ uit de toegankelijke naam te verwijderen en alleen ‘menu’ te gebruiken in combinatie met ‘aria-expanded’, aangezien dit al de juiste functionaliteit en status van de component weergeeft.

[Utility Navigation: What It Is and How to Design It](https://www.nngroup.com/articles/utility-navigation/)
Dit artikel bespreekt het belang van utility-navigatie, zoals functies voor contact, aanmelden, delen en printen, die secundaire acties ondersteunen. Hoewel de meeste functies van utility-navigatie van secundaire aard zijn, zijn ze belangrijk voor gebruikers in bepaalde situaties.

[The Current State of Homepage and Category Navigation UX](https://baymard.com/blog/ecommerce-navigation-best-practice)
Enkele interessante bevindingen: 
- Uit tests bleek dat wanneer de huidige toplevel-scope niet werd gemarkeerd in de hoofdnavigatie, het onnodig moeilijk was voor deelnemers om te bepalen waar ze zich in de sitestructuur bevonden. Zelfs bij perfect geïmplementeerde breadcrumb navigation vertrouwen sommige gebruikers nog steeds op de hoofdnavigatie om snel te begrijpen waar ze zijn of om te bevestigen dat ze zich op de juiste plek bevinden.
- Onduidelijke navigatie is één van de belangrijkste redenen voor het verlaten van een website. Er wordt aanbevolen om het menu te verbeteren. Dit kun je doen door klikbare headers te gebruiken in je multilevelmenu, ervoor te zorgen dat gebruikers duidelijk weten waar ze zich in je menu bevinden en een kleine vertraging toe te voegen wanneer de gebruiker de muis over de drop-down items in de hoofdnavigatie beweegt. 

[Universal Navigation: Connecting Subsites to Main Sites - NN/g](https://www.nngroup.com/articles/universal-navigation/)
Dit artikel bespreekt het belang van universele navigatie op grote websites met meerdere subsites, waarbij het gebruikers in staat stelt snel naar de hoofdpagina of andere gerelateerde sites te navigeren zonder de navigatie van de subsite te verstoren. En er worden ontwerpprincipes gedeeld om universele navigatie gebruiksvriendelijk te maken.

[Navigation - Yaale ](https://usability.yale.edu/web-accessibility/articles/navigation)
Dit artikel biedt richtlijnen voor het ontwerpen van navigatiemenu’s, zoals het gebruik van zichtbare koppen, toegankelijkheidspraktijken, en het implementeren van broodkruimels, koppen, en skip-links om de navigatie voor alle gebruikers te verbeteren. 

[Menu Structure - W3C](https://www.w3.org/WAI/tutorials/menus/structure/)
Informatie over hoe je een menustructuur toegankelijk opbouwt, met nadruk op het gebruik van semantische opmaak om de navigatie aanpasbaar te maken voor verschillende apparaten en hulpmiddelen voor toegankelijkheid.

[How to set up an accessible page structure and navigation - pope.tech](https://blog.pope.tech/2023/01/31/how-to-set-up-an-accessible-page-structure-and-navigation/)
Dit artikel bespreekt het belang van een toegankelijke paginstructuur en navigatie voor websites en geeft tips.

**Comment by Rozerinay** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-containers-header--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=1222-39598&node-type=canvas&t=zWlEYT9GU14eDwIT-11)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/header)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="952" alt="image" src="https://github.com/user-attachments/assets/489f70d7-2267-44d3-9e58-106b19119c04">

<img width="314" alt="image" src="https://github.com/user-attachments/assets/5191a561-721d-4800-808b-a8cae0fc3ff6">


For a sub-brand
<img width="952" alt="image" src="https://github.com/user-attachments/assets/7593a3d5-0da9-4492-bcde-db1e7529e893">

<img width="309" alt="image" src="https://github.com/user-attachments/assets/38df9038-8cde-455b-85d0-aa5f9e4117ce">


With app name
<img width="952" alt="image" src="https://github.com/user-attachments/assets/dc0f133d-c36a-420f-925b-576dc89d5865">

<img width="315" alt="image" src="https://github.com/user-attachments/assets/dc7e61e5-babc-4f27-8040-24a747d8e983">


With menu button
<img width="952" alt="image" src="https://github.com/user-attachments/assets/32fc3bff-6c5e-40d8-acd4-c1286a992ea3">

<img width="314" alt="image" src="https://github.com/user-attachments/assets/ee9c8004-457f-4e88-a040-64ecbc61cc88">


With links
Use a [Page Menu](https://designsystem.amsterdam/?path=/docs/components-navigation-page-menu--docs) to add links. A Page Menu in the Header should not wrap.
<img width="952" alt="image" src="https://github.com/user-attachments/assets/6978b4ec-f00e-4125-bb56-22fffc1ea5d5">

<img width="309" alt="image" src="https://github.com/user-attachments/assets/d57f3a7a-1eda-4762-961e-bf41e85b52c1">


With links and menu button
<img width="952" alt="image" src="https://github.com/user-attachments/assets/e644c7d2-9a27-49ad-9476-f97f7026c56f">
<img width="313" alt="image" src="https://github.com/user-attachments/assets/f4496cee-f6f7-4a6b-9a04-cee5946169eb">


With app name and menu button
<img width="952" alt="image" src="https://github.com/user-attachments/assets/b875685d-bfca-4320-a6ae-9139d163b705">

<img width="314" alt="image" src="https://github.com/user-attachments/assets/ceca2954-a17e-4f24-b8d1-0fe02f505d21">


With app name, links and menu button
<img width="952" alt="image" src="https://github.com/user-attachments/assets/136abe59-dbcb-407e-bf58-ee82f16cbd4e">

<img width="315" alt="image" src="https://github.com/user-attachments/assets/1a8529cf-d2bb-4606-8793-560e66fc152e">


---

### Notities
- Onder de naam: Header.
- Kan bestaan uit een logo, applicatienaam, links en een menubutton. 
- In de menubutton is het megamenu te vinden.
- Logo is standaard links uitgelijnd en rechts zijn de acties te vinden.
- Responsive gedrag: acties worden ondergebracht in de menubutton.
- Richtlijnen beschikbaar (Storybook).

**Comment by Rozerinay** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/story/css-page-elements-header--default)
- [Figma - Huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=12209-72381&node-type=frame&t=7yTsgdgmFyz7JLhS-11)
- [Github](https://github.com/nl-design-system/denhaag/tree/main/components/Header)

---

### Screenshots / Varianten

**Storybook**
<img width="2233" alt="image" src="https://github.com/user-attachments/assets/4131dba7-6cfa-4e32-b5e2-ee218111519f">

**Figma**
Logged out
<img width="768" alt="image" src="https://github.com/user-attachments/assets/f229f1e0-e91a-414d-8158-e5db8f50214f">

<img width="300" alt="image" src="https://github.com/user-attachments/assets/6f5aae1f-890b-4cf5-8d82-7514c6e6cf6f"><br>

Logged in
<img width="768" alt="image" src="https://github.com/user-attachments/assets/cdcb18a8-52c3-4972-b3ce-659dc9e09d12">

<img width="300" alt="image" src="https://github.com/user-attachments/assets/0aaa523d-7125-4765-9f7c-1005fa0230d8">

---

### Notities

- Onder de naam: Header.
- Kan bestaan uit een logo, link, icon, button en menubutton.
- In de menubutton is het megamenu te vinden.
- Logo is standaard links uitgelijnd en rechts zijn de acties te vinden.
- Responsive gedrag: acties worden ondergebracht in de button.
- Richtlijnen beschikbaar (Figma).

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/67.0.0/components/layout/header)
- [Storybook](https://storybook.dso-toolkit.nl/67.0.0/?path=/story/core-header--default&viewMode=story)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/header)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="980" alt="image" src="https://github.com/user-attachments/assets/6a77bce3-dac3-459a-945d-1da234097952">

User Home Active
<img width="980" alt="image" src="https://github.com/user-attachments/assets/2aa841f2-5fc4-45f6-9a6c-1cf3e5409f1f">

With Button To Help
<img width="980" alt="image" src="https://github.com/user-attachments/assets/3bcd3284-59c4-4774-827c-00dc85ce950d">

With Label
<img width="980" alt="image" src="https://github.com/user-attachments/assets/a9b777ed-26bd-4081-b5bd-452dc4e41f76">

With Label And Label URL
<img width="980" alt="image" src="https://github.com/user-attachments/assets/ca299bd5-76bb-4c2a-9e13-f4bd59153d4e">

With Label And Ribbon
<img width="980" alt="image" src="https://github.com/user-attachments/assets/bf3896ee-ef5d-4734-b9cf-5529422c4b85">

With Link To Help
<img width="980" alt="image" src="https://github.com/user-attachments/assets/555c73f4-8470-40e6-87be-08684a65281a">

With Logo Url And Label and Label Url
<img width="980" alt="image" src="https://github.com/user-attachments/assets/f1a41132-542b-4495-a865-7210110b4314">

With Ribbon
<img width="980" alt="image" src="https://github.com/user-attachments/assets/2d60dcf8-c83f-4144-91bf-20230ed4ae9a">

Responsive gedrag
<img width="360" alt="image" src="https://github.com/user-attachments/assets/9fbab9ad-4892-4bb0-94d8-8d6843914146">

<img width="360" alt="image" src="https://github.com/user-attachments/assets/cc0e7d1f-f664-419b-84fd-37a87d9a44bd">


---

### Notities

- Onder de naam: Header.
- Kan bestaan uit een logo, link en button.
- Logo is standaard links uitgelijnd en rechts zijn de acties te vinden. Hieronder staat de navbar.
- Bij veel links in de navbar worden de laatste links ondergebracht in een 'Meer' button.
- Er kan worden gekozen voor een compacte header: hiermee verdwijnt de navbar.
- Responsive gedrag: acties worden ondergebracht in de button. In dit menu verschijnen de acties op een Non Modal Dialog.

**Comment by Rozerinay** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/header)

---

### Screenshots / Varianten

**Documentatie website**
<img width="1335" alt="image" src="https://github.com/user-attachments/assets/e89ee5e3-d05e-4344-8195-94e5907dce1e">

Responsive gedrag

<img width="265" alt="image" src="https://github.com/user-attachments/assets/e64c3b28-c21b-4f89-be41-c0678fdd34d1">

---

### Notities

- Onder de naam: Header.
- Bestaat uit een logo en naam van de website.
- Logo is standaard midden uitgelijnd.
- Responsive gedrag: acties worden ondergebracht in de button. In dit menu verschijnen de acties op een Non Modal Dialog.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## OpenGemeenten

[Pattern Lab](https://patternlab.lenteveld.nl/?p=organisms-OpenGemeentenSite-PageHeader)

---

### Screenshots / Varianten

**Pattern Lab**
<img width="1293" alt="image" src="https://github.com/user-attachments/assets/04ae99ae-dc56-4d00-aaa9-3c74e1f21d5e">

<img width="1313" alt="image" src="https://github.com/user-attachments/assets/ce4cb80e-81eb-41cf-99c7-5c2293120951">

Responsive gedrag
<img width="364" alt="image" src="https://github.com/user-attachments/assets/f1515e80-d2a7-4abe-ac4c-38d4b74ab242">

<img width="364" alt="image" src="https://github.com/user-attachments/assets/4e688349-0d30-44fd-8676-d92f335390c6"><br>

<img width="364" alt="image" src="https://github.com/user-attachments/assets/0a33e6cf-e92f-49e1-bbf6-2ba2a97d5e66">

---
### Notities

- Onder de naam: Page Header.
- Kan bestaan uit een logo, links en een zoekbalk.
- Logo is standaard links uitgelijnd en rechts zijn de acties te vinden.
- Bij het aanklikken van de zoekbalk worden deze bovenop de acties geplaatst.
- Responsive gedrag: acties worden ondergebracht in een button. 
- Het menu en de zoekbalk kunnen niet tegelijkertijd worden gebruikt op small containers.

**Comment by Rozerinay** [comment]
## Rotterdam

- [Figma](https://www.figma.com/design/iWuQsxelJzXVfCge87Og81/RODS-component?node-id=1-3044&node-type=canvas&t=THUIlBW63uIsnbwy-11)
- [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/rotterdam-nlds-parent-wicket/rotterdam-nlds-wicket/src/main/java/nl/rotterdam/nl_design_system/wicket/components/page_header)

---

### Screenshots / Varianten

**Figma**
<img width="1486" alt="image" src="https://github.com/user-attachments/assets/d81e8c4e-1143-4bea-aa09-e03a6281d335">

---

### Notities

- Onder de naam: Main.
- Kan bestaan uit een top bar, logo, links en een menu button.
- Logo is standaard links uitgelijnd en rechts zijn de acties te vinden. Hierboven staat de top bar. 
- Responsive gedrag: acties worden ondergebracht in een button.

**Comment by Rozerinay** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/componenten/header/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-header--documentatie)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=46-528&node-type=canvas&t=YeOAN3FsqhN17qxZ-11)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/header)

---

### Screenshots / Varianten

**Storybook**

<img width="967" alt="image" src="https://github.com/user-attachments/assets/d80a20d4-e2aa-4644-8221-fae204b6ef23">

Responsive gedrag
<img width="303" alt="image" src="https://github.com/user-attachments/assets/bfc1cc96-f842-4794-a2e5-b2f59256ea07">

**Figma**
<img width="1359" alt="image" src="https://github.com/user-attachments/assets/dd883571-a8a8-4cec-8e43-e1903f5c0f03">


---

### Notities

- Onder de naam: Header.
- Bestaat uit een logo en naam van de website.
- Logo is standaard midden uitgelijnd.
- Responsive gedrag: logo en naam van de website verkleinen.

**Comment by Rozerinay** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-page-header--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/page-header)

---

### Screenshots / Varianten

**Storybook**

<img width="695" alt="image" src="https://github.com/user-attachments/assets/3c5e0584-9147-4724-95a4-a5f3c363b1e3">

Responsive gedrag
<img width="270" alt="image" src="https://github.com/user-attachments/assets/b5afdc08-0313-4ecd-96b8-f5085fdc9a33">


---

### Notities

- Onder de naam: Page Header.
- Bevat in het voorbeeld enkel de children 'Header content'.
- Kan bestaan uit een logo, navigatie, zoekveld, breadcrumb navigation, melding en cookie banner.
- Richtlijnen beschikbaar (Storybook).

**Comment by Rozerinay** [comment]
## Rijksoverheid

[Rijkshuisstijl](https://www.rijkshuisstijl.nl/publiek/modules/product/DigitalStyleGuide/default/index.aspx?ItemId=10824)

---

## Screenshots / Varianten

De nieuwe Header van de Rijksoverheid. 
![image](https://github.com/user-attachments/assets/71da3ba4-f8a0-4083-8f43-a9fffeb3748c)

Met kruimelpad
![image](https://github.com/user-attachments/assets/5c13abf9-f93d-4235-9fae-be854d1a9b29)

Uitklapmenu
![image](https://github.com/user-attachments/assets/08d761ee-220c-4582-b77c-ea5b3a65e436)

---

### Notities

- Onder de naam: Navigatie.
- Kan bestaan uit een logo, naam van platform, links, kruimelpad, zoekbutton en een menu button.
- In de menubutton is het megamenu te vinden.
- Logo is standaard midden uitgelijnd. Hieronder staan de links of het kruimelpad, samen met de zoekbutton. 
- Responsive gedrag: bij default worden acties worden ondergebracht in een button. Bij de kruimelpad variant is er sprake van horizontale scroll. Zoekbutton blijft altijd rechts uitgelijnd. 
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Page Header op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=page+header)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Page Header op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=page+header)
